Src/Fido2/AttestationFormat/Apple.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Asn1;
+using Fido2NetLib.Objects;
+using PeterO.Cbor;
+
+namespace Fido2NetLib.AttestationFormat
+{
+
+    internal class Apple : AttestationFormat
+    {
+        private readonly IMetadataService _metadataService;
+        public Apple(CBORObject attStmt, byte[] authenticatorData, byte[] clientDataHash, IMetadataService metadataService)
+            : base(attStmt, authenticatorData, clientDataHash)
+        {
+            _metadataService = metadataService;
+        }
+
+        public override void Verify()
+        {
+            // Reference: https://developer.apple.com/documentation/devicecheck/validating_apps_that_connect_to_your_server
+
+            // Verify that attStmt is valid CBOR conforming to the syntax defined above and
+            // perform CBOR decoding on it to extract the contained fields.
+            if (0 == attStmt.Keys.Count || 0 == attStmt.Values.Count)
+                throw new Fido2VerificationException("Attestation format apple must have attestation statement");
+
+            if (null == Alg || true != Alg.IsNumber)
+                throw new Fido2VerificationException("Invalid apple attestation algorithm");
+
+            // Verify that x5c is present
+            if (null == X5c || CBORType.Array != X5c.Type || 0 == X5c.Count || null != EcdaaKeyId)
+                throw new Fido2VerificationException("Malformed x5c array in apple attestation statement");
+
+            // Validate certificate lifetimes
+            var enumerator = X5c.Values.GetEnumerator();
+            while (enumerator.MoveNext())
+            {
+                if (null == enumerator || null == enumerator.Current

```

Comment:
`enumerator` will never be `null`, wil it?

Reply:
This was also copied, but I'm fixing.